src/networkapi.jl

```diff
@@ -1481,5 +1481,158 @@ function validate(rs::ReactionSystem, info::String = "")
     validated
 end
 
+"""
+    iscomplexbalanced(rs::ReactionSystem, rates::Vector)
+
+Constructively compute whether a network will have complex-balanced equilibrium
+solutions, following the method in [this paper](https://link.springer.com/article/10.1007/s10910-015-0498-2#Sec3). Accepts a map of rates [k1 => 1.0, k2 => 2.0,...]k2
+"""
+
+using Combinatorics, LinearAlgebra
+
+function iscomplexbalanced(rs::ReactionSystem, rates::Dict{Any, Float64}) 
```

Comment:
This should work for Vector/tuple mappings too as we commonly suggest them for users, along with specifying `Symbols` for the first component. You can use `symmap_to_varmap` to convert `Symbol` mappings to symbolic mappings, see https://docs.sciml.ai/Catalyst/stable/api/catalyst_api/#Catalyst.symmap_to_varmap.

Reply:
I would just not type `rates` for simplicity, but in the docstring mention the input options (vectors, tuples, dict mappings as accepted by ModelingToolkit for specifying parameter values).

Reply:
Inside the function you are welcome to convert the inputted mapping to a dictionary for ease of use if you want (and needed). It might make sense to just have three dispatches
```julia
function iscomplexbalanced(rs::ReactionSystem, parametermap::Vector)
     pdict = Dict(parametermap)
     iscomplexbalanced(rs, pdict)
end

function iscomplexbalanced(rs::ReactionSystem, parametermap::Tuple)
     pdict = Dict(parametermap)
     iscomplexbalanced(rs, pdict)
end

function iscomplexbalanced(rs::ReactionSystem, parametermap::Dict)
    your existing code
end

iscomplexbalanced(rs::ReactionSystem, parametermap) = error("parametermap is not in an allowable format.)
```
or similar.